format/format.go

```diff
@@ -44,19 +46,37 @@ func MustAst(x interface{}) []byte {
 // encountered, a default location will be set on the AST node.
 func Ast(x interface{}) (formatted []byte, err error) {
 
+	wildcards := map[string]struct{}{}
+	wildcardNames := map[string]string{}
+	wildcardCounter := 0
+
 	ast.WalkNodes(x, func(x ast.Node) bool {
-		if b, ok := x.(ast.Body); ok {
-			if len(b) == 0 {
+		switch n := x.(type) {
+		case ast.Body:
+			if len(n) == 0 {
 				return false
 			}
+		case *ast.Term:
+			if v, ok := n.Value.(ast.Var); ok {
+				if v.IsWildcard() {
+					str := string(v)
+					if _, seen := wildcards[str]; !seen {
+						wildcards[str] = struct{}{}
+					} else if !strings.HasPrefix(wildcardNames[str], "__VAR_") {
+						wildcardNames[str] = fmt.Sprintf("__VAR_%d__", wildcardCounter)
+						wildcardCounter++
+					}
+				}
+			}
 		}
```

Comment:
I feel like this logic could benefit from it's own function however I'd have to spend a bit of time thinking about how to best structure it to avoid multiple passes while setting the default location. Maybe a comment would be enough to explain what's going on here.